src/book/02-projects/01-nubots/03-build-system.mdx

```diff
@@ -5,3 +5,265 @@ title: Build System
+
+The `generic` target is useful if you would like to build and run code on your local machine.
+
+The `nuc7i7bnh` target should be used for building and running code on the robots.
+
+The `target` command will download (or, if needed, build) the Docker image for the specified target and then mark that target as active.
+
+## Continuous Integration
+
+NUbots uses [Buildkite](buildkite.com) for its continuous integration pipeline. The Buildkite server will issue build jobs to one of our build clients (the Macs in the lab) as new jobs become available.
+
+The build flow that the NUbots uses a pull request system. Developers branch off of `master`, creating a new branch named `<last name>/<purpose of branch>`. Once development on the branch is complete, the developer makes a pull request and a job is issued to a Buildkite client. Each pull request must be reviewed by another member of the team and all build checks on the Buildkite client must pass before the pull request can be approved and merged into `master`.
+
+## The Configuration System
+
+The NUbots configuration system is based on a hierarchical [YAML](https://yaml.org/) file system.
+
+The hierarchical structure is as follows
+
+```text
+config/
+├── nugus1/
+│   └── ExampleModule.yaml
+├── nugus2/
+│   └── ExampleModule.yaml
+├── robocup/
+│   └── ExampleModule.yaml
+└── ExampleModule.yaml
+```
+
+The file `config/ExampleMoldule.yaml` must exit for every module. This file defines all of the configuration for the module. `config/nugus1/ExampleModule.yaml` is an optional file and allows you to provide robot-specific overrides for specific configuration values, in this case it would provide overrides for the `nugus1` robot. Similarly, `config/robocup/ExampleModule.yaml` allows for the `robocup` role to further override configuration values.
+
+It is intended that all configuration values exist in the default configuration file and that only a subset of configuration values appear in the robot-specific and role-specific override files.
+
+The robot-specific override is intended to allow configuration values to be tweaked to account a slight variation in the robots hardware. For instance, the cameras on each robot have different serial numbers.
+
+The role-specific override is intended to change a modules behaviour based on which role is running. For instance, disabling the transmission of certain NUsight2 traffic while playing a competition game.
+
+## The Script System
+
+The script system follows a similar structure as the configuration system. However, default scripts are moved to a platform type folder and the overriding system is not used as it doesn't make much sense to try and merge two scripts.
+
+In the script system, we first check for a robot-specific script. If a robot-specific script does not exist, then we check for a platform-specific script.
```

Comment:
Do we ever check for a generic one?

Reply:
We only check those 2 for scripts. The "generic" one is the platform-specific one, because "generic" for a darwin is not the same as "generic" for a nugus